index.js

```diff
@@ -70,40 +70,50 @@ function parseDependencies (source, root) {
  * output transformed source.
  */
 function transformSource (runner, config, source, map, callback) {
-  var child = execFile(
+  var child = spawn(
     runner.file,
     runner.arguments.concat(
       runnerPath,
       ioDelimiter,
       config.engine
     ),
-    { timeout: config.timeoutMs },
-    function (error, stdout, stderr) {
-      // Output is delimited to filter out unwanted warnings or other output
-      // that we don't want in our files.
+    { stdio: ['pipe', 'pipe', process.stderr] }
+  )
+  if (config.timeoutMs) {
+    var cancelTimeout = setTimeout(function () {
+      child.kill()
+    }, config.timeoutMs)
+  }
+
+  var dataBuffers = []
+  child.stdout.on('data', function (data) {
+    // Output is delimited to filter out unwanted warnings or other output
+    // that we don't want in our files.
+    dataBuffers.push(data)
+  })
+
+  child.on('close', function (code) {
+    if (code === 0) {
       var sourceRegex = new RegExp(ioDelimiter + '([\\s\\S]+)' + ioDelimiter)
-      var matches = stdout.match(sourceRegex)
+      var matches = dataBuffers.join('').match(sourceRegex)
       var transformedSource = matches && matches[1]
-      if (error && error.signal === 'SIGTERM') {
-        if (config.timeoutMs) {
-          callback(new Error(
-            'rails-erb-loader took longer than the specified ' + config.timeoutMs +
-            'ms timeout'
-          ))
-        } else {
-          callback(error)
-        }
-      } else {
-        callback(error, transformedSource, map)
-      }
+      if (config.timeoutMs) { cancelTimeout() }
+      callback(null, transformedSource, map)
+    } else if (child.killed) {
+      callback(new Error(
+        'rails-erb-loader took longer than the specified ' + config.timeoutMs +
+        'ms timeout'
+      ))
+    } else {
+      callback(new Error('rails-erb-loader failed with code: ' + code))
     }
-  )
+  })
+
   child.stdin.on('error', function (error) {
-    if (error.code === 'EPIPE') {
-      // When the `runner` command is not found, stdin will not be open.
-      // Attemping to write then causes an EPIPE error. Ignore this because the
-      // `exec` callback gives a more meaningful error that we show to the user.
-    } else {
+    // When the `runner` command is not found, stdin will not be open.
+    // Attemping to write then causes an EPIPE error. Ignore this because the
+    // `exec` callback gives a more meaningful error that we show to the user.
```

Comment:
This comment may be wrong now since we're no longer called `exec` (in truth it was already stale because we changed to `execFile`). Either way, if this logic is correct it I suspect it would be the `child.on('error'` callback that gives the better report.

Reply:
Looks like the `child.on('error')` gets an `ENOENT` if the runner is not found, so we could probably loose this listener entirely. That being said, it doesn't hurt to keep it here either so removing the comment and if statement around the `console.warn` seems like the better solution for future debugging